src/OSVRTrackedDevice.cpp

```diff
@@ -848,21 +848,34 @@ std::string OSVRTrackedDevice::GetStringTrackedDeviceProperty(vr::ETrackedDevice
     return default_value;
 }
 
-void OSVRTrackedDevice::HmdTrackerCallback(void* userdata, const OSVR_TimeValue*, const OSVR_PoseReport* report)
+void OSVRTrackedDevice::HmdTrackerCallback(void* userdata, const OSVR_TimeValue* timeval, const OSVR_PoseReport* report)
 {
     if (!userdata)
         return;
 
     auto* self = static_cast<OSVRTrackedDevice*>(userdata);
 
+    // Get angular velocity in correct format for SteamVR
+    OSVR_TimeValue timeval2;
+    OSVR_AngularVelocityState state;
+    OSVR_VelocityState velocitystate;
+
+    osvrGetVelocityState(self->trackerInterface_.get(), &timeval2, &velocitystate);
+
+    double dt = state.dt;
+    Eigen::Quaterniond poserotation = osvr::util::fromQuat(report->pose.rotation);
+    Eigen::Quaterniond angvel_incrementalRotation = osvr::util::fromQuat(velocitystate.angularVelocity.incrementalRotation);
+    angvel_incrementalRotation = poserotation.inverse() * angvel_incrementalRotation * poserotation;
```

Comment:
The intention is to have the HMD be the frame of reference of the angular velocity.
They do not cancel each other because rotation composition and quaternion multiplication are not commutative.